src/schema/comments.ts

```diff
@@ -608,7 +609,15 @@ export const resolvers: IResolvers<any, Context> = {
               .andWhere(`${builder.alias}.postId = :postId`, {
                 postId: args.postId,
               })
-              .andWhere(`${builder.alias}.parentId is null`);
+              .andWhere(`${builder.alias}.parentId is null`)
+              // Only show banned comments if the user is the author of the comment
+              .andWhere(
+                new Brackets((qb) => {
+                  qb.where(`${builder.alias}.userId = :userId`, {
+                    userId: ctx.userId,
+                  }).orWhere(`${builder.alias}.banned = false`);
+                }),
+              );
```

Comment:
@idoshamun @capJavert thought? I filter out banned comments on API side, except if you're the author of the comment, so you'll still be see your own comments even if they're banned.

Reply:
Nvm, think I got it in 4dc0ca52ae4c2b9841425c62d04233a908fb5206

Reply:
Should we add this query logic to a shared place so we can reuse in graphorm and in here? Just in case we change it at some point

Reply:
I can maybe extract the following into a separate function?

```ts
new Brackets((qb) => {
  qb.where(`${childAlias}.userId = :userId`, {
    userId: ctx.userId,
  }).orWhere(`${childAlias}.banned = false`);
})
```

Reply:
Extracted to own function, see
https://github.com/dailydotdev/daily-api/blob/966ec5bbb870dea037306181b06f3763eb9d30d0/src/common/vordr.ts#L87-L92